lib/krb5/dcache.c

```diff
@@ -779,9 +779,10 @@ dcc_get_default_name(krb5_context context, char **str)
                                        "libdefaults", "default_cc_collection",
                                        NULL);
 
-    /* What if def_cc_colname does not start with DIR:?  We tolerate it. */
-    return _krb5_expand_default_cc_name(context, def_cc_colname,
-					str);
+    /* [libdefaults] default_cc_collection is for testing */
+    if (strncmp(def_cc_colname, "DIR:", sizeof("DIR:") - 1))
+        def_cc_colname = KRB5_DEFAULT_CCNAME_DIR;
+    return _krb5_expand_default_cc_name(context, def_cc_colname, str);
 }
```

Comment:
As a general rule the `get_default_name()` methods of ccaches should always return just their hard-coded default.  The `DIR` ccache does this instead so that `tests/kdc/check-cc.in` can test it.  It'd be nice to generalize this somewhat.